variables.tf

```
@@ -40,11 +40,18 @@ variable "launch_template_version" {
 }
 
 variable "associate_public_ip_address" {
-  type        = bool
-  description = "Associate a public IP address with an instance in a VPC"
+  type = bool
+  # https://stackoverflow.com/a/76808361
+  description = "Associate a public IP address with an instance in a VPC. If a network_interface id is specified, this can only be false."
```

@gbarna-bd this is a good clarification, but let's include it into the description. 

```suggestion
  type        = bool
  description = "Associate a public IP address with an instance in a VPC. If `network_interface_id` is specified, this can only be `false` (see here for more info: https://stackoverflow.com/a/76808361)."
```

---

Done. Thanks.